### PR TITLE
Set up simulation server

### PR DESCRIPTION
**Pull request by blauks**
I haven't tested this yet, but it runs!

### CODE REVIEW

File: simulation_server/src/main/kotlin/projectcows/Application.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import io.ktor.server.application.*
+import io.ktor.server.engine.*
+import io.ktor.server.netty.*
+import projectcows.plugins.*
+import io.ktor.server.plugins.contentnegotiation.*
+import io.ktor.serialization.gson.*
+import java.text.DateFormat
+
+fun main() {
+    embeddedServer(Netty, port = 8069, host = "0.0.0.0") {
```
Comment (AndreasWintherMoen) [inline simulation_server/src/main/kotlin/projectcows/Application.kt:12]: Would be really nice to load port from an environment file, but I know it's a bit tricky in Java/Kotlin so it's okay if you don't bother.

File: simulation_server/src/main/kotlin/projectcows/enums/GameState.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.cows.game.enums
```
Comment (AndreasWintherMoen) [inline simulation_server/src/main/kotlin/projectcows/enums/GameState.kt:1]: Do we need this file??

**Review by AndreasWintherMoen — APPROVED** [review]
Cool! Looks good to me (see minor remarks in comments). We have to remove the simulation module from the client, but we can do that in a separate PR tomorrow :)

File: simulation_server/src/main/kotlin/projectcows/Application.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package projectcows
```
[the remark this answers is not in the document]
Comment (blauks) [reply simulation_server/src/main/kotlin/projectcows/Application.kt:1]: Didn't really think about that when I created the project, can fix it

File: simulation_server/src/main/kotlin/projectcows/enums/GameState.kt (hunk 2 above)
Comment (blauks) [reply simulation_server/src/main/kotlin/projectcows/enums/GameState.kt:1]: Nope!

**Comment by Nicolaad** [comment]
I attempted to test the RoundSimulatior, by attempting to print the eventlog. However, I faced issues with the map dependency.

You can try by running
` val parsedFile = File("roundSimulation.json").readText()
        val roundSimulation = RoundSimulationDeserializer.deserialize(parsedFile)
        var simulator = RoundSimulator()

        val gson = Gson()
        val simulated = simulator.simulate(roundSimulation.towerList, roundSimulation.unitList)
        val writeFile = File("roundSImulatorOutput.json").writeText(gson.toJson(simulated))
        print(gson.toJson(simulated.eventLog))
`

However, you will need to get the path to simulated, as well as link to the file in client

**Comment by blauks** [comment]
> I attempted to test the RoundSimulatior, by attempting to print the eventlog. However, I faced issues with the map dependency.
> 
> You can try by running (in the test)
> 
> ```
> val parsedFile = File("roundSimulation.json").readText()
> val roundSimulation = RoundSimulationDeserializer.deserialize(parsedFile)
> var simulator = RoundSimulator()
> 
> val gson = Gson()
> val simulated = simulator.simulate(roundSimulation.towerList, roundSimulation.unitList)
> val writeFile = File("roundSimulatorOutput.json").writeText(gson.toJson(simulated))
> print(gson.toJson(simulated.eventLog))
> ```
> 
> However, you will need to get the path to simulated, as well as link to the file in client

You have to send in the path to the simulator.simulate method (as a list of int lists), as the SS will not have any information about the game. Does that fix it?

**Comment by Nicolaad** [comment]
nvm, given that you have tested it, it seems reasonable.

**Review by Nicolaad — APPROVED** [review]

**Pull request merged**